ecr/resolver.go

```diff
@@ -161,21 +159,37 @@ func (r *ecrResolver) Resolve(ctx context.Context, ref string) (string, ocispec.
 		WithField("batchGetImageOutput", batchGetImageOutput).
 		Debug("ecr.resolver.resolve")
 
-	var ecrImage *ecr.Image
 	if len(batchGetImageOutput.Images) == 0 {
 		return "", ocispec.Descriptor{}, reference.ErrInvalid
 	}
-	ecrImage = batchGetImageOutput.Images[0]
+	ecrImage := batchGetImageOutput.Images[0]
+
 	mediaType := parseImageManifestMediaType(ctx, aws.StringValue(ecrImage.ImageManifest))
 	log.G(ctx).
 		WithField("ref", ref).
-		WithField("media type", mediaType).
+		WithField("mediaType", mediaType).
 		Debug("ecr.resolver.resolve")
+	// check resolved image's mediaType, it should be one of the specified in
+	// the request.
+	for i, accepted := range aws.StringValueSlice(batchGetImageInput.AcceptedMediaTypes) {
+		if mediaType == accepted {
+			break
+		}
+		if i+1 == len(batchGetImageInput.AcceptedMediaTypes) {
+			return "", ocispec.Descriptor{}, errors.Wrap(errdefs.ErrFailedPrecondition, "resolved mediaType not in accepted types")
+		}
+	}
```

Comment:
I don't think we should do this check.  The caller of the resolver can do such an assertion; it's possible that it'll make a different decision based on the response.

Reply:
Good point - is the expected pattern that you'd use a handler to filter out undesired content types? If so, should the resolver provide a predefined handler that filters down to the list of "supported" types in lieu of this check?

Reply:
I think we can provide a handler, but we don't have to.  It's not an error if the media type doesn't match the types we pass as accepted in the call; it just means that ECR didn't have content that matched the requested media type and couldn't transform it, but is still a valid response.  The caller (of this library) needs to be able to decide if it knows how to handle that content.

Reply:
Okay, I've removed the logic that bails on the unknown types. This might be better in-scope when considering adding or updating options to the resolver.

Reply:
It looks like the code is still here in [`53a3003` (#13)](https://github.com/awslabs/amazon-ecr-containerd-resolver/pull/13/commits/53a30030c3d5ccf097cb965f5946f2365b949413#diff-c82d190de1354947418a3d1d2ee93a83R38-R42); can you remove it?

Reply:
It's been replaced with the logging alternative in what was this commit.